modulus/models/registry.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pkg_resources
+from typing import List, Union
+
+# This model registry follows conventions similar to fsspec,
+# https://github.com/fsspec/filesystem_spec/blob/master/fsspec/registry.py#L62C2-L62C2
+# Tutorial on entrypoints: https://amir.rachum.com/blog/2017/07/28/python-entry-points/
+class ModelRegistry:
+    _instance = None
+    _model_registry = {}
+
+    def __new__(cls):
+        if not cls._instance:
+            cls._instance = super(ModelRegistry, cls).__new__(cls)
```

Comment:
What is the benefit of this style of singleton pattern vs the Borg pattern we have for `DistributedManager`? If there is one pattern that is clearly better, we should switch both classes to use that pattern.

Some interesting discussion here: https://stackoverflow.com/questions/1318406/why-is-the-borg-pattern-better-than-the-singleton-pattern-in-python

Reply:
No preference here so switched to borg.